### PR TITLE
Add tests for ChunkStreamingParser and ChunkDownloaderFactory

### PR DESCRIPTION
**Pull request by sfc-gh-ext-simba-lf**
Regarding issue 113

The PR adds tests for ChunkStreamingParser and ChunkDownloaderFactory to increase code coverage

### CODE REVIEW

File: Snowflake.Data.Tests/ChunkDownloaderFactoryTest.cs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        [TearDown]
+        public void AfterTest()
+        {
+            SFConfiguration.Instance().UseV2ChunkDownloader = UseV2ChunkDownloader; // Return to default version
+            SFConfiguration.Instance().ChunkDownloaderVersion = ChunkDownloaderVersionDefault; // Return to default version
+        }
+
+        [Test]
+        [Ignore("ChunkDownloaderFactoryTest")]
+        public void ChunkDownloaderFactoryTestDone()
```
Comment (sfc-gh-mhofman) [inline Snowflake.Data.Tests/ChunkDownloaderFactoryTest.cs:29]: What for are those ignored empty tests?

File: Snowflake.Data.Tests/ChunkStreamingParserTest.cs [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        }
+
+        [Test]
+        public void TestParsingNonJsonArrayAndOneArray()
+        {
+            // Create a sample data using non-array data and an array
+            string data = "[ \"1\", \"1.234\", \"abcde\", [\"2\", \"5.678\", \"fghi\"] ]";
+            IChunkParser parser = getParser(data);
+
+            SFResultChunk chunk = new SFResultChunk(new string[1, 3]);
```
Comment (sfc-gh-mhofman) [inline Snowflake.Data.Tests/ChunkStreamingParserTest.cs:103]: if 2nd dimension would be higher that the test fails which means that parser didn't throw an exception.
Is this expected behaviour?

File: Snowflake.Data.Tests/ChunkStreamingParserTest.cs [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+            // Should throw an error when parsing non-array and array data together
+            Assert.ThrowsAsync<IndexOutOfRangeException>(async () => await parser.ParseChunk(chunk));
+        }
+
+        [Test]
+        public void TestParsingJsonObjectChunk()
```
Comment (sfc-gh-mhofman) [inline Snowflake.Data.Tests/ChunkStreamingParserTest.cs:110]: As above I would provide some info about expected behaviour:
TestThrowingExceptionForJsonObjectElementsInAnArray ?

File: Snowflake.Data.Tests/ChunkDownloaderFactoryTest.cs (hunk 1 above)
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data.Tests/ChunkDownloaderFactoryTest.cs:29]: I think those ignored tests were added to track the test progress on github actions: https://github.com/snowflakedb/snowflake-connector-net/pull/464/commits/cf1abe2c940b8b62d8eb97a5837cb31059bfb2d1

File: Snowflake.Data.Tests/ChunkStreamingParserTest.cs [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+        {
+            // Create a sample data using JSON objects instead
+            string data = "[ {\"1\", \"1.234\", \"abcde\"},  {\"2\", \"5.678\", \"fghi\"} ]";
+            IChunkParser parser = getParser(data);
+
+            SFResultChunk chunk = new SFResultChunk(new string[2, 3]);
+
+            // Should throw an error when parsing a JSON object
+            Assert.ThrowsAsync<SnowflakeDbException>(async () => await parser.ParseChunk(chunk));
```
[the remark this answers is not in the document]
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data.Tests/ChunkStreamingParserTest.cs]: Sounds good, I'll add that

File: Snowflake.Data.Tests/ChunkStreamingParserTest.cs (hunk 3 above)
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data.Tests/ChunkStreamingParserTest.cs]: Ok

File: Snowflake.Data.Tests/ChunkStreamingParserTest.cs (hunk 2 above)
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data.Tests/ChunkStreamingParserTest.cs:103]: I'm honestly not sure if that's the expected behavior. The parser class was created before we started and I couldn't find any tests/docs about it

I'm assuming it was never meant to parse non-array data based on this: https://github.com/snowflakedb/snowflake-connector-net/blob/master/Snowflake.Data/Core/SFBlockingChunkDownloaderV3.cs#L216C5-L216C5

**Review by sfc-gh-mhofman — APPROVED** [review]

**Review by sfc-gh-igarish — APPROVED** [review]
LGTM

**Pull request merged**